lib/openstack_api/dataclasses.py

```diff
@@ -59,5 +59,19 @@ class EmailQueryParams:
     required_email_property: str
     valid_search_queries: List[str]
     valid_search_queries_no_project: List[str]
-    search_api: Any
-    object_type: str
+
+
+@dataclass
+class QueryParams:
+    """
+    Contains the data needed for running OpenstackQuery.search_resource
+    :param query_preset: The query to use when searching
+    :param properties_to_select: The properties to select and display from the found resource
+    :param group_by: Property to group the results by - can be empty for no grouping
+    :param get_html: Whether the result should be in html format or not
+    """
+
+    query_preset: str
+    properties_to_select: List[str]
```

Comment:
If we have a list of known properties can we use an enum to constrain the list of available values. Currently we have to look at the implementation each time to see the supported params 
https://docs.python.org/3/library/enum.html#using-automatic-values

Reply:
This is a bit tricky as this is a general dataclass for all resources. It would be possible to add them in the actions but it would be a long list that's tedious to update.

Reply:
Ah I thought it was a fairly consistent subset with a few bespoke ones rather than bespoke to each class :+1: